development/components/database/objectmodel.md

```diff
@@ -1,16 +1,56 @@
 ---
 title: ObjectModel class
+useMermaid: true
 ---
 
 # The ObjectModel class
 
-When needing to dive deep, you have to use the ObjectModel class. This is the main object of PrestaShop’s object model. It can be overridden… with precaution.
+## Introduction
 
-It is an Active Record kind of class (see: [Active record pattern](https://en.wikipedia.org/wiki/Active_record_pattern)). The table attributes or view attributes of PrestaShop’s database are encapsulated in this class. Therefore, the class is tied to a database record. After the object has been instantiated, a new record is added to the database. Each object retrieves its data from the database; when an object is updated, the record to which it is tied is updated as well. The class implements accessors for each attribute.
+ObjectModel class is one of the main pillar of PrestaShop's legacy core. While a complete migration to Symfony/Doctrine entities is planned in the roadmap, ObjectModel will still remain present and available in our software for a while.
 
-## Defining the model
+ObjectModel is the Data Access Layer for PrestaShop, implemented following the Active Record pattern. The Data Access Layer (DAL) is a part (with the Database Abstraction Layer - DBAL) of the Object Relational Mapping (ORM) legacy system for PrestaShop.
 
-You must use the `$definition` static variable in order to define the model.
+Read more about Object relation mapping (ORM), Database abstraction layer (DBAL), Data access layer (DAL), and Active Record :
+
+- [Object relational mapping (ORM)](https://en.wikipedia.org/wiki/Object%E2%80%93relational_mapping)
+- [Database abstraction layer (DBAL)](https://en.wikipedia.org/wiki/Database_abstraction_layer)
+- [Data access layer (DAL)](https://en.wikipedia.org/wiki/Data_access_layer)
+- [Active record pattern](https://en.wikipedia.org/wiki/Active_record_pattern)
+
+A class extending ObjectModel class is tied to a database table. Its static attribute (`$definition`) is representing the model. 
+
+Its instances are tied to database records. The ObjectModel class provides accessors for each attribute defined in `$definition`. 
+
+When instancied with an `$id` passed to the class constructor, the attributes are retrieved in the related database (using the `$id` as the primary key to find the table record). 
+
+When the `save()` method is called, ObjectModel will ask the DBAL to **insert** or **update** the object to database, depending if the `$id` is known or not in the ObjectModel.
+
+ObjectModel is also in charge of deleting an object in database (with its `delete()` method).
```

Comment:
```suggestion
ObjectModel is also in charge of deleting an object in database (with its `delete()` method).
```

ObjectModel is full CRUD. I think we could say this instead of talking about specific responsibilities 👍